ch08/lambda/hello-delta-python/deploy/hello.tf

```diff
@@ -0,0 +1,120 @@
+#
+# This file contains the main resources necessary to configure and deploy the Lambda
+#
+
+locals {
+  lambda_file = "../target/lambda_function.zip"
+  layer_file  = "../target/lambda_layer.zip"
+}
+
+resource "aws_lambda_function" "hello" {
+  description      = "A simple Lambda that says hello with Delta metadata"
+  function_name    = "hello-delta-python"
+  filename         = local.lambda_file
+  source_code_hash = filebase64sha256(local.lambda_file)
+  role             = aws_iam_role.iam_for_lambda.arn
+  memory_size      = 512
+  handler          = "lambda_function.lambda_handler"
+  runtime          = "python3.10"
+  timeout          = 60
+  # Using the AWS SDK Pandas layer as well, since that provides pyarrow and pandas
+  layers = [
+    aws_lambda_layer_version.python.arn,
+    "arn:aws:lambda:us-west-2:336392948345:layer:AWSSDKPandas-Python310:8"
+  ]
+
+  environment {
+    variables = {
+      RUST_LOG  = "deltalake=info"
+      TABLE_URL = "s3://${aws_s3_bucket.data.bucket}/COVID-19_NYT"
+    }
+  }
+}
+
+resource "aws_s3_object" "layer" {
+  bucket = aws_s3_bucket.data.id
+  key    = "lambda_layer.zip"
+  etag   = filebase64sha256(local.layer_file)
+  source = local.layer_file
+}
+resource "aws_lambda_layer_version" "python" {
+  layer_name          = "python_with_delta"
+  s3_bucket           = aws_s3_bucket.data.id
+  s3_key              = aws_s3_object.layer.key
+  s3_object_version   = aws_s3_object.layer.version_id
+  compatible_runtimes = ["python3.10"]
+}
+
+resource "aws_lambda_function_url" "hello" {
+  function_name      = aws_lambda_function.hello.function_name
+  authorization_type = "NONE"
+}
+
+output "lambda_function_url" {
+  value = aws_lambda_function_url.hello.function_url
+}
+
+resource "aws_s3_bucket" "data" {
+  # Generating a random bucket name to ensure readers don't have conflicts!
+  bucket = "deltalake-examples-${random_id.random.hex}"
+}
+
+# Upload the test Delta Table
+resource "aws_s3_object" "covid19" {
+  for_each = fileset("../../../../datasets/COVID-19_NYT/", "**")
+  bucket   = aws_s3_bucket.data.id
+  key      = "COVID-19_NYT/${each.value}"
+  source   = "../../../../datasets/COVID-19_NYT/${each.value}"
+}
+
+resource "random_id" "random" {
+  byte_length = 8
+}
+
+data "aws_iam_policy_document" "assume_role" {
```

Comment:
this is awesome to have in the examples. I now feel like I need to modify the code for chapter 14 to just have the tf plan for s3 sql grants, bucket setup, tags, and policy definitions. +100;